src/Raven.Studio/typescript/viewmodels/shell/UpgradeModal.tsx

```diff
@@ -3,41 +3,103 @@ import { clusterSelectors } from "components/common/shell/clusterSlice";
+    const canActivate = asyncGetConfigurationSettings.result?.CanActivate;
+    const isReplaceLicenseEnabled = canActivate && isClusterAdminOrClusterNode;
+
+    const licenseStatus = useAppSelector(licenseSelectors.status);
+    
+    const registerLicense = () => {
+        registration.showRegistrationDialog(licenseStatus, false, true);
+
+        // Remove tabindex from bs5 modal for bs3 to work
+        const upgradeModalElement = document.querySelector("#bs5-modal [role='dialog']");
+        if (upgradeModalElement && upgradeModalElement.hasAttribute("tabindex")) {
+            upgradeModalElement.removeAttribute("tabindex");
+        }
+    };
 
     if (!upgradeRequired || !latestVersion || !productVersion) {
         return null;
     }
 
+    const allowDismissUntilUtc = moment.utc(upgradeRequired.AllowDismissUntil);
+
     return (
-        <Modal
-            show
-            contentClassName="modal-border bulge-warning"
-            size="lg"
-        >
-            <Modal.Body className="vstack gap-3 position-relative justify-content-center">
-                <div className="d-flex justify-content-center mb-3">
+        <Modal show contentClassName="modal-border bulge-warning" size="lg">
+            <Modal.Header
+                className="vstack gap-4"
+                closeButton={upgradeRequired.AllowDismiss}
+                onCloseClick={props.close}
+            >
+                <div className="d-flex justify-content-center">
                     <img src={upgradeLicenseImg} alt="Upgrade license" width="120" />
                 </div>
                 <h3 className="text-warning text-center mb-0">It&apos;s time to upgrade!</h3>
+            </Modal.Header>
+            <Modal.Body>
                 <p className="text-center mb-0">
                     Your server is running version <strong>{productVersion}</strong> while the latest version is{" "}
                     <strong>{latestVersion}</strong>.
                     <br />
                     In order to continue using RavenDB please upgrade your server to the latest available version.
+                    {upgradeRequired.AllowDismiss && upgradeRequired.AllowDismissUntil && (
+                        <>
+                            <br />
+                            <br />
+                            <span className="text-muted">
+                                You can dismiss this message until {allowDismissUntilUtc.format(dateFormat)} UTC
+                                <PopoverWithHoverWrapper
+                                    message={`${allowDismissUntilUtc.local().format(dateFormat)} your local time`}
+                                >
+                                    <Icon icon="info" color="info" margin="ms-1" />
+                                </PopoverWithHoverWrapper>. 
+                                Afterwards the RavenDB studio will be blocked.
```

Comment:
I do not like how this is rendered, should be in two lines without a dot `.` 
Maybe also change text to `RavenDB Studio will be blocked afterwards` 
- make RavenDB Studio bold
- make 'blocked' red and bold

Reply:
done